File: services/erp.ts
import api from './api';

export interface Branch {
  id: string;
  name: string;
  location?: string;
  createdAt: string;
}

export interface InventoryItem {
  id: string;
  branchId?: string;
  name: string;
  category: string;
  serialNumber?: string;
  status: string;
  purchasePrice: number;
  sellingPrice: number;
  purchaseDate?: string;
  supplier?: string;
  warrantyMonths: number;
  quantity: number;
  lowStockThreshold: number;
  createdAt: string;
}

export interface Technician {
  id: string;
  branchId?: string;
  fullName: string;
  phone: string;
  cnic?: string;
  assignedArea?: string;
  vehicle?: string;
  availability: string;
  completedJobs: number;
  avgResolutionTimeMins: number;
  createdAt: string;
}

export const erpService = {
  // Branches
  getBranches: async (): Promise<Branch[]> => {
    const response = await api.get('/../../v2/erp/branches');
    return response.data;
  },
  
  createBranch: async (branchData: any): Promise<Branch> => {
    const response = await api.post('/../../v2/erp/branches', branchData);
    return response.data;
  },

  // Inventory
  getInventory: async (): Promise<InventoryItem[]> => {
    const response = await api.get('/../../v2/erp/inventory');
    return response.data;
  },
  
  createInventoryItem: async (itemData: any): Promise<InventoryItem> => {
    const response = await api.post('/../../v2/erp/inventory', itemData);
    return response.data;
  },
  
  updateInventoryItem: async (id: string, itemData: any): Promise<InventoryItem> => {
    const response = await api.put(`/../../v2/erp/inventory/${id}`, itemData);
    return response.data;
  },
  
  deleteInventoryItem: async (id: string): Promise<InventoryItem> => {
    const response = await api.delete(`/../../v2/erp/inventory/${id}`);
    return response.data;
  },

  // Technicians
  getTechnicians: async (): Promise<Technician[]> => {
    const response = await api.get('/../../v2/erp/technicians');
    return response.data;
  },
  
  createTechnician: async (techData: any): Promise<Technician> => {
    const response = await api.post('/../../v2/erp/technicians', techData);
    return response.data;
  },
  
  updateTechnician: async (id: string, techData: any): Promise<Technician> => {
    const response = await api.put(`/../../v2/erp/technicians/${id}`, techData);
    return response.data;
  },
};

File: backend/app/models/erp.py
import uuid
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, Float
from backend.app.database.session import Base
from datetime import datetime

class Branch(Base):
    __tablename__ = "branches"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    tenant_id = Column(String(36), default="friends_network", nullable=True)
    name = Column(String(100), unique=True, nullable=False, index=True)
    location = Column(String(200), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)

class InventoryItem(Base):
    __tablename__ = "inventory_items"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    tenant_id = Column(String(36), default="friends_network", nullable=True)
    branch_id = Column(String(36), ForeignKey("branches.id"), nullable=True)
    name = Column(String(150), nullable=False)
    category = Column(String(100), nullable=False) # "Routers" | "ONUs" | "Fiber Cable" | "Splitters" | "UPS"
    serial_number = Column(String(100), unique=True, nullable=True, index=True)
    status = Column(String(50), default="In Stock") # "In Stock" | "Assigned" | "Defective"
    purchase_price = Column(Integer, default=0)
    selling_price = Column(Integer, default=0)
    purchase_date = Column(String(10), nullable=True) # "YYYY-MM-DD"
    supplier = Column(String(100), nullable=True)
    warranty_months = Column(Integer, default=12)
    quantity = Column(Integer, default=1)
    low_stock_threshold = Column(Integer, default=5)
    created_at = Column(DateTime, default=datetime.utcnow)

class Technician(Base):
    __tablename__ = "technicians"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    tenant_id = Column(String(36), default="friends_network", nullable=True)
    branch_id = Column(String(36), ForeignKey("branches.id"), nullable=True)
    full_name = Column(String(100), nullable=False)
    phone = Column(String(50), nullable=False)
    cnic = Column(String(50), nullable=True)
    assigned_area = Column(String(100), nullable=True)
    vehicle = Column(String(100), nullable=True)
    availability = Column(String(50), default="Available") # "Available" | "On Duty" | "Off Duty"
    completed_jobs = Column(Integer, default=0)
    avg_resolution_time_mins = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
